hypertrace-ingester/src/test/java/org/hypertrace/migration/MigrationTestRpc.java

```diff
@@ -0,0 +1,335 @@
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.typesafe.config.ConfigFactory;
+import io.jaegertracing.api_v2.JaegerSpanInternalModel.Span;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+import java.util.stream.Stream;
+import org.hypertrace.core.datamodel.RawSpan;
+import org.hypertrace.core.semantic.convention.constants.error.OTelErrorSemanticConventions;
+import org.hypertrace.core.semantic.convention.constants.rpc.OTelRpcSemanticConventions;
+import org.hypertrace.core.span.constants.RawSpanConstants;
+import org.hypertrace.core.spannormalizer.jaeger.JaegerSpanNormalizer;
+import org.hypertrace.semantic.convention.utils.rpc.RpcSemanticConventionUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class MigrationTestRpc {
+
+  private final Random random = new Random();
+  private JaegerSpanNormalizer normalizer;
+
+  @BeforeEach
+  public void setup()
+      throws SecurityException, NoSuchFieldException, IllegalArgumentException,
+          IllegalAccessException {
+    // Since JaegerToRawSpanConverter is a singleton, we need to reset it for unit tests to
+    // recreate the instance.
+    Field instance = JaegerSpanNormalizer.class.getDeclaredField("INSTANCE");
+    instance.setAccessible(true);
+    instance.set(null, null);
+
+    // Create a JaegerSpanNormaliser
+    String tenantId = "tenant-" + this.random.nextLong();
+    Map<String, Object> configs = Map.of("processor", Map.of("defaultTenantId", tenantId));
+    this.normalizer = JaegerSpanNormalizer.get(ConfigFactory.parseMap(configs));
+  }
+
+  @Test
+  public void testGrpcFields() throws Exception {
+
+    Map<String, String> tagsMap =
+        new HashMap<>() {
+          {
+            put(RawSpanConstants.getValue(GRPC_ERROR_MESSAGE), "Some error message");
+            put(RawSpanConstants.getValue(CENSUS_RESPONSE_STATUS_CODE), "12");
+            put(RawSpanConstants.getValue(GRPC_STATUS_CODE), "13");
+            put(RawSpanConstants.getValue(CENSUS_RESPONSE_CENSUS_STATUS_CODE), "14");
+            put(
+                RawSpanConstants.getValue(CENSUS_RESPONSE_STATUS_MESSAGE),
+                "CENSUS_RESPONSE_STATUS_MESSAGE");
+            put(RawSpanConstants.getValue(ENVOY_GRPC_STATUS_MESSAGE), "ENVOY_GRPC_STATUS_MESSAGE");
+            put(RawSpanConstants.getValue(GRPC_REQUEST_BODY), "some grpc request body");
+            put(RawSpanConstants.getValue(GRPC_RESPONSE_BODY), "some grpc response body");
+          }
+        };
+
+    Span span = createSpanFromTags(tagsMap);
+    RawSpan rawSpan = normalizer.convert("tenant-key", span);
+
+    assertAll(
```

Comment:
All the exposed methods are not covered. So, lets add here,
- `getGrpcAuthority`
- `getGrpcURI`
- `getGrpcURL`
- `getGrpcUserAgent`
- `getGrpcXforwardedFor`
- `getRpcMethod`, `getRpcOpration`, `getRpcService` (if they are public, and used in enricher.)
- `getGrpcRequestMetadataPath` if this is used somewhere.

Reply:
In this pr lets add test for the apis which are used in ht trace-enricher only

Reply:
Do you want theses tests in rpcsemantictest file or in migrationrpctest file ?

Reply:
I think better to keep it in migration, since its easier to compare parity with normalizer code

Reply:
Except last 2 all are already present in rpcsemantictest file

Reply:
Added all the tests in rpctest file only

Reply:
@sarthak77 we are still missing for - getGrpcXForwardedFor. It should be a copy of the user-agent test.

Reply:
Its already there. 
` 
@Test
  public void testGetGrpcXForwardedFor() {
    Event event = mock(Event.class);
    when(event.getAttributes())
        .thenReturn(
            Attributes.newBuilder()
                .setAttributeMap(
                    Map.of(
                        OTEL_SPAN_TAG_RPC_SYSTEM.getValue(),
                        AttributeValue.newBuilder().setValue("grpc").build(),
                        RPC_REQUEST_METADATA_X_FORWARDED_FOR.getValue(),
                        AttributeValue.newBuilder().setValue("198.12.34.1").build()))
                .build());
    assertEquals("198.12.34.1", RpcSemanticConventionUtils.getGrpcXForwardedFor(event).get());
  }
`